### PR TITLE
fix(ImageGrid): 3-side and 4-side doesn't always display the right number of images 

### PR DESCRIPTION
**Pull request by chienleng**
## Description

💁‍
* filter side images based on layout type
* update styles

## Related issues

💪
https://app.shortcut.com/qantasaccommodation/story/79437/image-gallery-standard-layout

## Looks like this

🤨
Screen Shot 2021-11-12 at 11 58 46 am

---

**PR check-list**

🚨 **IMPORTANT:** Please review the [guidelines for contributing](./CONTRIBUTING.md) to this repository first! 👀

- [ ] You've requested a review from at least ONE person. Preferably a roo-ui maintainer ([Philip](https://github.com/philipwindeyer), [Eric](https://github.com/eneo5541), [Mike](https://github.com/memcc), [Marcel](https://github.com/Anthropic), [Sam](https://github.com/SamuSan) [Long](https://github.com/KieraDOG)) or a Roo UI working-group human.
  - Note: If your PR has not been reviewed within two days of requesting a review, feel free to seek reviews from other [team members of Qantas Hotels](https://github.com/orgs/hooroo/people).
- [ ] Your final merge commit message is in the format `<type>(<scope>): <description>` as per [the contributing guide](https://github.com/hooroo/roo-ui/blob/master/.github/CONTRIBUTING.md#commit-naming)
  - Example: `fix(dependencies): minor version bump to resolve security vulnerability in set-value`

### CODE REVIEW

**Review by philipwindeyer — APPROVED** [review]
🚀

File: src/components/ImageGrid/__snapshots__/ImageGrid.test.js.snap [hunk 1]
```diff
@@ -46,6 +46,12 @@ exports[`<ImageGrid /> displays 3 side layout 1`] = `
   background-color: transparent;
 }
 
+@media (max-width:) {
```
Comment (mrm007) [inline src/components/ImageGrid/__snapshots__/ImageGrid.test.js.snap:49]: This doesn't look right.

File: src/components/ImageGrid/primitives.js [hunk 2]
```diff
@@ -13,12 +13,16 @@ export const StyledGrid = styled(Flex)`
 
 export const StyledMainImage = styled(NakedButton)`
   flex-grow: 1;
+
+  @media (max-width: ${themeGet('breakpoints.1')}) {
```
Comment (mrm007) [inline src/components/ImageGrid/primitives.js:17]: I can see why this isn't showing up in the snapshot. [`mountWithTheme`](https://github.com/hooroo/roo-ui/blob/bc525e90098f008e436ec3f3a69df7090874550d/src/testUtils.js#L6) from `testUtils` doesn't wrap the tree with a `ThemeProvider` so `themeGet` can't find it.

File: src/components/ImageGrid/__snapshots__/ImageGrid.test.js.snap (hunk 1 above)
Comment (chienleng) [reply src/components/ImageGrid/__snapshots__/ImageGrid.test.js.snap]: updated test and snapshot

File: src/components/ImageGrid/primitives.js (hunk 2 above)
Comment (chienleng) [reply src/components/ImageGrid/primitives.js:17]: updated test and snapshot

Comment (mrm007) [reply src/components/ImageGrid/primitives.js:17]: Nice one!

**Pull request merged**